Makefile

```diff
@@ -18,41 +18,45 @@ virtualenv: $(PYTHON)
 $(PYTHON):
 	$(VIRTUALENV) $(VENV)
 
-install-dev: $(INSTALL_STAMP) $(DEV_STAMP)
+install-dev: $(INSTALL_STAMP) $(DEV_STAMP) ## Install development dependencies
 $(DEV_STAMP): $(PYTHON) dev-requirements.txt
 	$(VENV)/bin/pip install -Ur dev-requirements.txt
 	touch $(DEV_STAMP)
 
 remove-install-stamp:
 	rm $(INSTALL_STAMP)
 
-update: remove-install-stamp install
+update: remove-install-stamp install ## Update the dependencies
 
-serve: install
+serve: install ## Run the ihatemoney server
+	@echo 'Running ihatemoney on http://localhost:5000'
 	$(PYTHON) -m ihatemoney.manage runserver
 
-test: $(DEV_STAMP)
+test: $(DEV_STAMP) ## Run the tests
 	$(VENV)/bin/tox
 
-release: $(DEV_STAMP)
+release: $(DEV_STAMP) ## Release a new version
 	$(VENV)/bin/fullrelease
 
-build-translations:
+build-translations: ## Build the translations
 	$(VENV)/bin/pybabel compile -d ihatemoney/translations
 
-update-translations:
+update-translations: ## Update the translations
 	$(VENV)/bin/pybabel extract --strip-comments --omit-header --no-location --mapping-file ihatemoney/babel.cfg -o ihatemoney/messages.pot ihatemoney
 	$(VENV)/bin/pybabel update -i ihatemoney/messages.pot -d ihatemoney/translations/
 
-create-database-revision:
+create-database-revision: ## Create a new database revision
 	@read -p "Please enter a message describing this revision: " rev_message; \
 	$(PYTHON) -m ihatemoney.manage db migrate -d ihatemoney/migrations -m "$${rev_message}"
 
-build-requirements:
+build-requirements: ## Freeze the requirements
 	$(VIRTUALENV) $(TEMPDIR)
 	$(TEMPDIR)/bin/pip install -U pip
 	$(TEMPDIR)/bin/pip install -Ue "."
 	$(TEMPDIR)/bin/pip freeze | grep -v -- '-e' > requirements.txt
 
-clean:
+clean: ## Clean the virtual environment
 	rm -rf .venv
+
+help: ## Show the help indications
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
```

Comment:
Nice trick :-)

Reply:
Thanks :) This one is not from me but I use it on all the projects I work on, very helpful to self-document the Makefile!